web/src/main/java/com/futurewei/alcor/web/entity/dataplane/GoalStateJsonNB.java

```diff
@@ -0,0 +1,94 @@
+package com.futurewei.alcor.web.entity.dataplane;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.futurewei.alcor.common.constants.Common;
+import com.futurewei.alcor.common.entity.CustomerResource;
+import com.futurewei.alcor.web.entity.securitygroup.SecurityGroupEntity;
+import com.futurewei.alcor.web.entity.subnet.SubnetEntity;
+import com.futurewei.alcor.web.entity.vpc.VpcEntity;
+
+public class GoalStateJsonNB extends CustomerResource {
+
+  private Common.ResourceType rsType;
+  private Common.OperationType opType;
+  boolean allOrNone = true;
+
+  @JsonProperty("ports")
+  private InternalPortEntityNB[] portStates;
+
+  @JsonProperty("vpcs")
+  private VpcEntity[] vpcs;
+
+  @JsonProperty("subnets")
+  private SubnetEntity[] subnets;
+
+  @JsonProperty("host_infos")
+  private HostInfoNB[] hostInfoNBs;
+
+  @JsonProperty("security_groups")
+  private SecurityGroupEntity[] securityGroupEntities;
+
+  public HostInfoNB[] getHostInfoNBs() {
+    return hostInfoNBs;
+  }
+
+  public void setHostInfoNBs(HostInfoNB[] hostInfoNBs) {
+    this.hostInfoNBs = hostInfoNBs;
+  }
+
+  public SecurityGroupEntity[] getSecurityGroupEntities() {
+    return securityGroupEntities;
+  }
+
+  public void setSecurityGroupEntities(SecurityGroupEntity[] securityGroupEntities) {
+    this.securityGroupEntities = securityGroupEntities;
+  }
+
+  public Common.ResourceType getRsType() {
+    return rsType;
```

Comment:
I think we could use an antation @Data, and remove all the setter and getter.

Reply:
An example is https://github.com/futurewei-cloud/alcor/blob/master/web/src/main/java/com/futurewei/alcor/web/entity/port/PortEntity.java